libfreerdp/core/peer.c

```diff
@@ -996,27 +1000,29 @@ static int peer_recv_callback_internal(rdpTransport* transport, wStream* s, void
 						                                         .deviceScaleFactor = 100 } };
 					if (!freerdp_settings_set_pointer_array(settings, FreeRDP_MonitorDefArray, 0,
 					                                        &primary))
-						ret = -2;
+						ret = STATE_RUN_FAILED;
 					else if (!freerdp_settings_set_uint32(settings, FreeRDP_MonitorCount, 1))
-						ret = -3;
+						ret = STATE_RUN_FAILED;
 				}
-				if (ret < -1)
+				if (state_run_failed(ret))
 				{
```

Comment:
Strange statement here, we do nothing if `state_run_failed(ret)`, perhaps the condition should be inverted with a `if (!state_run_failed(ret)) { if (!display_convert_rdp_m...` ?

Reply:
tried to avoid another indentation level here, but you´re right.